packages/botonic-plugin-nlu/src/model-handler.ts

```diff
@@ -0,0 +1,80 @@
+import { default as fetch } from 'node-fetch'
+import { Vocabulary } from '@botonic/nlu/dist/types'
+import { Language } from '@botonic/nlu/dist/language'
+import { Preprocessor } from '@botonic/nlu/dist/preprocessor'
+
+import {
+  BotonicPluginNLUOptions,
+  ModelInformation,
+  PreprocessingOptions,
+  Result,
+} from './types'
+import { getModelInfoForEnv } from './environment-utils'
+import { detectLang, inputToTensor, predictionToIntent } from './prediction'
+
+// @ts-ignore
+global.fetch = fetch
+
+export class ModelHandler {
+  private languages: Language[]
+  private modelInfo: ModelInformation = {}
+
+  constructor(options: BotonicPluginNLUOptions) {
+    // @ts-ignore
+    return (async () => {
+      this.languages = Object.keys(options) as Language[]
+      await this.loadModelInformation(options)
+      return this
+    })()
+  }
+
+  initPreprocessor(
+    options: PreprocessingOptions,
+    vocabulary: Vocabulary,
+    maxSeqLen: number
+  ): Preprocessor {
+    const preprocessor = new Preprocessor()
+    if (options.tokenizer) preprocessor.tokenizer = options.tokenizer
+    if (options.normalizer) preprocessor.normalizer = options.normalizer
+    if (options.stemmer) preprocessor.stemmer = options.stemmer
+    preprocessor.vocabulary = vocabulary
+    preprocessor.maxSeqLen = maxSeqLen
+    return preprocessor
+  }
+
+  async loadModelInformation(options: BotonicPluginNLUOptions): Promise<this> {
+    for (const lang of this.languages) {
+      this.modelInfo[lang] = {}
+      this.modelInfo[lang].language = lang
+      // Prepare model information promises to be waited in parallel
+      const { model, modelData } = getModelInfoForEnv(lang) as any
+      this.modelInfo[lang].model = model
```

Comment:
model and & modelData are Promises. you shouldn't assign them to non-promise fields
I suggest to 
1) create a local var.
```
const promises : {[lang: string] : ModelInfoPromises
```
2 feed it
```
promises[lang] =  getModelInfoForEnv(lang)

3 use it
```
 for (const [lang, res] of Object.entries(this.modelInfo)) {
      this.modelInfo[lang].model = await promises[lang].model
      this.modelInfo[lang].modelData = (await promises[lang].modelData).data
```

We are already waiting in parallel since the promises are launched in parallel.

Reply:
now it's looking cleaner, well seen 💯